06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Implementation of the inverted index app.
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * @desc Initializes indexedFiles object
+   */
+  constructor() {
+    this.indexedFiles = {};
+  }
+
+  /**
+   * @desc readFile: Reads the contents of file as text
+   * @param {file} file - the file to be read
+   * @param {string} fileName - the name of the file to be read
+   * @return {promise} - promise object to be acted on if file
+   * reading is succesful or not
+   */
+  readFile(file, fileName) {
+    this.file = file;
```

Comment:
Eslint kept flagging that a class method should have 'this' in it. That was the solution I came up with to quiet the nagging.